src/dvc_objects/db.py

```diff
@@ -90,8 +97,6 @@ def add_bytes(self, oid: str, data: Union[bytes, BinaryIO]) -> None:
             size = cast("Optional[int]", getattr(fobj, "size", None))
 
         path = self.oid_to_path(oid)
-        parent = self.fs.path.parent(path)
-        self.makedirs(parent)
```

Comment:
hmm, but we do need this for correctness sake, right?

Reply:
not sure what you mean. db needs to be initialized before you start adding stuff into it.

Reply:
The underlying filesystem may still call makedirs, as is the case in LocalFileSystem. Object Storages are noop already.

Reply:
API also does not make that much sense now, since you cannot `add` without initializing it.

Reply:
This is fine as a convention. This will make a comeback once `add()` can handle batches.

Reply:
We could also use a flag like odb._initialized and call `self.init()` in `add` for now, so it is only called once. Might do that in a followup.

Reply:
Yeah, let me do just that actually...

Reply:
https://github.com/iterative/dvc-objects/pull/147